### PR TITLE
Cleanup deprecated apis

### PR DESCRIPTION
**Pull request by yjhmelody**
Clean up some deprecated apis for pallets :
- Some apis that should be removed after july.
- Some apis that are deprecated more than one year.

------
Polkadot address: yjhmelody

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3856994

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3856965

**Comment by yjhmelody** [comment]
@bkchr

**Review by bkchr — APPROVED** [review]

File: substrate/frame/support/src/storage/migration.rs [hunk 1]
```diff
@@ -254,11 +120,7 @@ pub fn put_storage_value<T: Encode>(module: &[u8], item: &[u8], hash: &[u8], val
 /// `hash`.
 #[deprecated = "Use `clear_storage_prefix` instead"]
```
[the remark this answers is not in the document]
Comment (juangirini) [reply substrate/frame/support/src/storage/migration.rs:121]: I guess that can also be deleted, as even if it didn't have the deprecation date noted is already 16 months old

**Review by juangirini — COMMENTED** [review]
This needs a prdoc https://github.com/paritytech/polkadot-sdk/blob/master/docs/CONTRIBUTING.md#documentation

Looping in @sacha-l and @Rashmirreddy for potential communication and documentation.

**Review by kianenigma — COMMENTED** [review]
Won't block this, but won't approve as-is either to suggest this guideline for these PRs: 

- In the PR description, make a list of items you are removing
- Link to the PR merge date when they got merged, and approximately which release they went into.

Hopefully this is not too much work, but it gives us a very yardstick to know what we are deleting.

All in all though, there is a lot of such code in the codebase. Thanks for attending to them!

**Comment by yjhmelody** [comment]
Ok, I'm closed.

**Pull request closed without merging**